### DISCUSSION TITLE
Integrate JobRunr into Play Framework

### DISCUSSION THREAD

**Discussion by tanin47** [jobrunr/jobrunr#1177] [Show and tell]
Here's the full write-up: https://tanin.nanakorn.com/integrate-jobrunr-into-play-framework/

The main complication are: 

1. Play Framework uses Guice as its injection framework
2. Play Framework is in Scala

So, it needs a bit of extra setup. I hope this is helpful.

I have a PlayFramework's template that integrates with JobRunr here: https://github.com/tanin47/playfast -- just in case you want to see a working example

**Comment by sideeffffect** [comment]
For others looking to use JobRunr from Scala, here's a working snippet:
 * https://github.com/jobrunr/jobrunr/discussions/127#discussioncomment-12461138

**Comment by auloin** [comment]
Hi @tanin47,

I've included your article in our docs as an example on how to integrate JobRunr in Scala and Play: https://www.jobrunr.io/en/documentation/getting-started/other/. Thanks for writing it, I've the feeling that a lot of people started using JobRunr thanks to you!

I also wanted to give you a quick heads-up. We've updated the Fluent API configuration so that processes only start after the call to `initialize`. See PR: https://github.com/jobrunr/jobrunr/pull/1566. This will be released in 8.7. You may need to adapt your setup which relied on earlier behaviour.

Side note: the recurring job gotcha in your article should no longer be an issue in JobRunr v8, with ahead of time scheduling.

**Reply by tanin47** [reply]
Thank you! I'll take a look.

**Reply by tanin47** [reply]
> the recurring job gotcha in your article should no longer be an issue

This is superb

I wonder if I can ask for this to be solved as well: `We've "unprivate" the background server member in order to support running all the pending background jobs in test. We'll show it in the later section.` -- it is for flushing the pending background jobs in test. A better mechanism is welcome as well.

**Reply by auloin** [reply]
@tanin47, yes this is also fixed by PR https://github.com/jobrunr/jobrunr/pull/1566. The `JobRunrConfigurationResult` will additionally expose a `getBackgroundJobServer` method, and a `getDashboardWebServer`.

**Reply by tanin47** [reply]
This is awesome!
